s3wrapper/s3.go

```
@@ -297,3 +297,34 @@ func (w *S3Wrapper) CopyAll(keys chan *ListOutput, source, dest string, delimite
 
 	return listOut
 }
+
+// ListBucketsInput is an empty struct, therefore
+// we can define it globally here
+var listBucketsParams *s3.ListBucketsInput
+
+// ListBuckets returns a list of bucket names and does a prefix
+// filter based on s3Uri (of the form s3://<bucket-prefix>)
+func (w *S3Wrapper) ListBuckets(s3Uri string) ([]string, error) {
+
+	bucket, _ := parseS3Uri(s3Uri)
+	results, err := w.svc.ListBuckets(listBucketsParams)
+	buckets := make([]string, 0, len(results.Buckets))
+	if err != nil {
+		return buckets, err
+	}
+
+	bucketRegex := regexp.MustCompile("^" + bucket)
```

probably not, should probably return a custom error